### PR TITLE
[gen_l10n] Escape quote characters in ARB files

### LINKED ISSUE #51787: [gen_l10n] Does not escape single or double quotes properly.
Arb file:
```
  "rallyLoginNoAccount": "Het jy nie 'n rekening nie?",
```

Generated getter:
```
  @override
  String get rallyLoginNoAccount => 'Het jy nie 'n rekening nie?'; // causes error
```

### PR DESCRIPTION
**Pull request by shihaohong**
## Description

When using the l10n tool, the following issue has surfaced:

Arb file:
```
  "rallyLoginNoAccount": "Het jy nie 'n rekening nie?",
```

Generated getter:
```
  @override
  String get rallyLoginNoAccount => 'Het jy nie 'n rekening nie?'; // causes error
```

This PR introduced proper escaping of quote characters. There was also a small change to the stocks app example to make sure to use `en.arb` as the template instead of `en_US.arb`

## Related Issues

Fixes https://github.com/flutter/flutter/issues/51787

## Tests

I added the following tests:
- Additions to the integration test ensuring that double and single quote characters are properly generated by the tool.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: dev/tools/localization/gen_l10n.dart [hunk 1]
```diff
@@ -163,6 +163,11 @@ String generateMethod(Message message, AppResourceBundle bundle) {
         messageValue = messageValue.replaceAll('{${placeholder.name}}', '\${${placeholder.name}}');
       }
     }
+
+    // Escape single and double quotes.
+    messageValue = messageValue.replaceAll("'", '\\\'');
```
Comment (HansMuller) [inline dev/tools/localization/gen_l10n.dart]: Shouldn't we be using generateString - https://github.com/flutter/flutter/blob/master/dev/tools/localization/localizations_utils.dart#L387

Comment (shihaohong) [reply dev/tools/localization/gen_l10n.dart]: That seems to be breaking integration tests. Investigating

Comment (shihaohong) [reply dev/tools/localization/gen_l10n.dart]: It seems to be generating raw strings for scenarios like:
```
  @override
  String helloOn(Object world, DateTime date, DateTime time) {
    final intl.DateFormat dateDateFormat = intl.DateFormat.yMd(_localeName);
    final String dateString = dateDateFormat.format(date);
    final intl.DateFormat timeDateFormat = intl.DateFormat.Hm(_localeName);
    final String timeString = timeDateFormat.format(time);

    return r"'Hello ${world} on ${dateString} at ${timeString}'";
  }
```

I could repurpose the generateString function to not do so

Comment (HansMuller) [reply dev/tools/localization/gen_l10n.dart]: Lets leave the PR as it is for now.

File: packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart [hunk 2]
```diff
@@ -286,6 +288,16 @@ void main() {
       "hello": {},
       "world": {}
     }
+  },
+
+  "singleQuote": "Flutter's amazing!",
+  "@singleQuote": {
+    "description": "Love Flutter!"
```
Comment (HansMuller) [inline packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart]: In most of the rest of this file, the message descriptions briefly describe what we're checking.

**Review by HansMuller — APPROVED** [review]

File: packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart (hunk 2 above)
Comment (shihaohong) [reply packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart]: Modified it to describe the test

**Pull request merged**